### ISSUE TITLE
When using watcher, the thread will automatically exit

### ISSUE THREAD

**Issue by yess1987** [etcd-cpp-apiv3/etcd-cpp-apiv3#140]
When using watcher, the thread will automatically exit

**Comment by sighingnow** [comment]
Could you please provide some minimal example code that I can reproduce this issue? 

Thanks!

**Comment by yess1987** [comment]
上下文：在一个进程中，启动一个线程，持续监听固定路径的watcher；添加watcher后，发现watcher自身启动的线程过个几秒中会自动退出了，导致一直无法收到事件
环境：centos stream9，etcd V3.5.3 ，etcd-cpp-APIv3最新版本
问题:
1)是否可以不用自己再去启动单独线程替代getchar();
2)watcher中启动的线程自动退出了，退出条件是啥

void watcher_cb(etcd::Response const & resp) {
    if (resp.error_code()) {
        //error
        //std::cout<<"[lib]watcher_cb error_code:"<<resp.error_code()<<" and error";
        //std::cout <<" [lib]msg:"<<resp.error_message()<<std::endl;
        common_loginfo(PRINT_WARN, CLUSTER_COMMON_LOG, "[lib]watcher_cb error_code:%s\n", resp.error_code());
        common_loginfo(PRINT_WARN, CLUSTER_COMMON_LOG, " [lib]msg:%s\n", resp.error_message());
    } else {
        /*
        if(resp.action() == "set") {
            std::cout<<"[lib]set value:"<<resp.value().as_string() <<"\n";
        } else if(resp.action() == "delete") {
            std::cout<<"[lib]set value:"<<resp.value().as_string() <<"\n";
        }*/
        //std::cout<<"[lib]set value:"<<resp.value().as_string() <<"\n";
        common_loginfo(PRINT_INFO, CLUSTER_COMMON_LOG, "[lib]watcher_cb,key:%s action:%s value:%s\n", test_watcher.key,test_watcher.action,test_watcher.value);
    }
}



void thread_func(const char *strl, const char *key)
{
    etcd::Watcher watcher(strl, key, watcher_cb, false);
    //getchar();
    while(1)
    {
        sleep(1);
     }
}
int kvdb_set_watcher(const char *strl, const char *key)
{
    std::thread test(thread_func, strl, key);
    test.detach();
    common_loginfo(PRINT_INFO, CLUSTER_COMMON_LOG, "[lib]thread:%d\n", std::this_thread::get_id());
    return 0;
}

**Comment by yess1987** [comment]
已经解决此问题

**Comment by sighingnow** [comment]
Closing as fixed.
